pkg/controllers/state/cluster.go

```diff
@@ -82,6 +83,17 @@ func NewCluster(clk clock.Clock, client client.Client) *Cluster {
 //
 //nolint:gocyclo
 func (c *Cluster) Synced(ctx context.Context) (synced bool) {
+	if c.unsyncedState.IsZero() {
```

Comment:
But we only want to set this unsyncedState one time and then measure the time difference between unsyncedState and syncedState. Isn't that what this is? If I set it every time when this function returns false then the time difference is always going to be zero.
```
defer func() {
		if synced {
			ClusterStateUnsyncedTimeSeconds.With(map[string]string{}).Set(0)
		} else {
			c.unsyncedStartTime = c.clock.Now()
			ClusterStateUnsyncedTimeSeconds.With(map[string]string{}).Set(time.Since(c.unsyncedStartTime).Seconds())
		}
	}()
```

Reply:
Seems like you could just move the check in the `defer` statement. I think what confused me was the fact that these things were placed in the code separately